contracts/Synthswap.sol

```diff
@@ -21,28 +25,117 @@ contract SynthSwap is ISynthSwap {
+        uint256 inputSynthAmount,
+        address destinationToken,
+        uint256 slippage
+    ) external payable override returns (uint) {
+
+        // Approve the Synthetix router to spend inputSynth.
+        IERC20 InputERC20 = IERC20(inputSynth);
+        uint synthBalance = InputERC20.balanceOf(address(this));
+        require(synthBalance >= inputSynthAmount, "insufficient synth balance");
+        InputERC20.approve(address(Synthetix), synthBalance);
+
+        // Swap inputSynth with sUSD by providing both the source and destination currency keys. 
+        Synthetix.exchangeWithTrackingForInitiator(
+            inputSynthCurrencyKey, // source currency key
+            synthBalance, // source amount
+            SUSD_CURRENCY_KEY, // destination currency key
+            volumeRewards, // volume rewards address
+            'KWENTA' // tracking code
+        );
+
+        // Approve the 1inch router to spend sUSD.
+        uint sUSDBalance = sUSD.balanceOf(address(this));
+        sUSD.approve(aggregationRouterV3, sUSDBalance);
+
+        // Swap sUSD with destinationToken
+        swapOnOneInch(address(sUSD), destinationToken, sUSDBalance, address(this), slippage);
+        uint amountReceived = IERC20(destinationToken).balanceOf(address(this));
+
+        sUSD.transfer(msg.sender, amountReceived);
+
+        emit SwapOutOf(msg.sender, amountReceived);
+        return amountReceived;
+    }
+
+    /**
+     * @notice Performs a token swap via 1inch (https://docs.1inch.io/api/quote-swap).
+     * @param fromTokenAddress contract address of a token to sell
+     * @param toTokenAddress contract address of a token to buy
+     * @param amount amount of a token to sell
+     * @param fromAddress address of a seller
+     * @param slippage limit of price slippage you are willing to accept in percentage [0-50]
+     */
+    function swapOnOneInch(
+        address fromTokenAddress,
+        address toTokenAddress,
+        uint256 amount,
+        address fromAddress,
+        uint256 slippage
+    ) internal {
+        bytes memory _data = abi.encodeWithSignature(
+            "swap(address,address,uint256,address,uint256)",
+            fromTokenAddress,
+            toTokenAddress,
+            amount,
+            fromAddress,
+            slippage
+        );
+        invoke(aggregationRouterV3, _data);
+    }
+
+    /**
+     * @notice Performs a generic transaction.
+     * @param _target The address for the transaction.
+     * @param _data The data of the transaction.
+     */
+    function invoke(address _target, bytes memory _data) internal returns (bytes memory) {
+        // External contracts can be compiled with different Solidity versions
+        // which can cause "revert without reason" when called through,
+        // for example, a standard IERC20 ABI compiled on the latest version.
+        // This low-level call avoids that issue.
+
+        (bool success, bytes memory returnData) = _target.delegatecall(_data);
```

Comment:
I think we should change this to `call`. `delegatecall` could introduce more implications we're not ready for.

Previously I had `delegatecall` so that we could pass `msg.value` through from the original caller (for ETH cases).

Reply:
Okay! The example I saw used `call()` but I kept it similar to what you had incase you had specific reasons for using `delegateCall()`.